src/read.rs

```diff
@@ -165,81 +166,81 @@ impl Description {
     pub fn sample_rate(&self) -> u32 {
         self.sample_rate
     }
-
 }
 
-
 impl BufFileReader {
-
     /// Attempts to open an audio `Reader` from the file at the specified `Path`.
     ///
     /// This function is a convenience wrapper around the `Reader::new` function.
     ///
     /// This function pays no attention to the `file_path`'s extension and instead attempts to read
     /// a supported `Format` via the file header.
     pub fn open<P>(file_path: P) -> Result<Self, ReadError>
-        where P: AsRef<std::path::Path>,
+    where
+        P: AsRef<std::path::Path>,
     {
         let path = file_path.as_ref();
-        let file = try!(std::fs::File::open(path));
+        let file = r#try!(std::fs::File::open(path));
```

Comment:
I've never seen this `r#` syntax in front of a `try!` macro invocation before! Is this a rustfmt bug I wonder? Either way, perhaps it's better to replace the use of the `try!` macro with the `?` operator these days.

Reply:
`r#try!``is needed because `try` is a keyword in edition 2018. `r#` is needed to signify that it refers to an identifier and not the keyword. For more details [here](https://doc.rust-lang.org/edition-guide/rust-2018/module-system/raw-identifiers.html). The `try!` macro is being deprecated separately from that and I think nightly should already warn about the deprecation.

Reply:
Yeah, I wasn't sure if I should use this syntax or `?`. So I just left as is after `cargo fix`. Will replace with `?` across all the changes

Reply:
I've replaced `r#try!` with `?`